File: my-folder/1619-path-crossing/solution.cpp
class Solution {
public:
    bool isPathCrossing(string path) {
        vector<pair<int,int>> s;
        int x = 0;
        int y = 0;
        s.push_back({x,y});
        for(char i : path){
            if(i == 'N') y++;
            else if(i == 'S') y--;
            else if(i == 'E') x--;
            else if(i == 'W') x++;
            s.push_back({x,y});
        }
        set<pair<int,int>> se(s.begin(),s.end());
        return s.size() != se.size();
    }
};

File: my-folder/1619-path-crossing/solution.py
class Solution:
    def isPathCrossing(self, path: str) -> bool:
        s={(0,0)}
        x_cor,y_cor=0,0
        for i in path:
            if i=='N':
                y_cor+=1
            if i=='S':
                y_cor-=1
            if i=='E':
                x_cor+=1
            if i=='W':
                x_cor-=1
            if (x_cor,y_cor) in s :
                    return True
            else : 
                s.add((x_cor,y_cor))
        return False
